pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+      __attribute__((unavailable));
+      
++ (instancetype)allocWithZone:
+     (struct _NSZone *)zone 
+     __attribute__((unavailable));
+     
++ (instancetype)demoObject 
+     __attribute__((swift_name("init()")));
+     
+@property (readonly) NSString *field;
+@end;
+
+__attribute__((objc_subclassing_restricted))
+@interface DemoDemoClazz : KotlinBase
+
+- (instancetype)init 
+     __attribute__((swift_name("init()"))) 
+     __attribute__((objc_designated_initializer));
+     
++ (instancetype)new 
+     __attribute__((availability(swift, unavailable, message="use object initializers instead")));
+     
+- (int64_t)memberFunctionP:(int32_t)p 
+     __attribute__((swift_name("memberFunction(p:)")));
+@end;
+
+__attribute__((objc_subclassing_restricted))
+@interface Demo : KotlinBase
+
++ (void)forIntegersB:(int8_t)b s:(int16_t)s i:(int32_t)i l:(int64_t)l 
+     __attribute__((swift_name("forIntegers(b:s:i:l:)")));
+     
++ (void)forFloatsF:(float)f d:(double)d 
+     __attribute__((swift_name("forFloats(f:d:)")));
+     
++ (NSString *)stringsStr:(NSString *)str 
+     __attribute__((swift_name("strings(str:)")));
+     
+@property (class, readonly) NSString *globalString;
+@end;
+
+NS_ASSUME_NONNULL_END
+```
+</div>
+
+The name `DemoDemoClazz` and `DemoDemoObject` are created for `DemoClazz` and `DemoObject` 
+Kotlin classes. The `Demo` prefix comes from the `-output` parameter
+of the `kotlinc` compiler and the Framework name. All global functions from Kotlin
+are turned into `Demo` class in Objective-C/Swift, where `Demo` is the Framework name again.
+You may notice, that Kotlin `String` and Objective-C `NSString` are mapped transparently.
+Similarly, `Unit` type from Kotlin is mapped to `void`. We see primitive types
+are mapped directly.
+
+You may want to see [Objective-C Interop](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md)
+documentation article to learn all other cases in detail.
+
+## Using the Code from Objective-C
+
+Let's call the Kotlin Framework from Objective-C. For that we create the `main.m` file with 
+the following content:
+
+<div class="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```obj-c 
+
+#import <Foundation/Foundation.h>
+#import <Demo/Demo.h>
+
+int main(int argc, const char * argv[]) {
+    
+    [[DemoDemoObject demoObject] field];
+    
+    DemoDemoClazz* clazz = [[ DemoDemoClazz alloc] init];
+    [clazz memberFunctionP:42];
+    
+    [Demo forIntegersB:1 s:1 i:3 l:4];
+   
+    NSString* ret = [Demo stringsStr:(@"That is string")];
+    return 0;
+}
+```
+</div>
+
+We call Kotlin classes directly from Objective-C code. Kotlin `object` has the class method 
+function `demoObject`, which allows us to get the only instance of the object and to call 
+`DemoObject` methods on it. 
+The standard pattern is used to create an instance of the `DemoClazz` class. We call
+the `[[ DemoDemoClazz alloc] init]` on Objective-C.
```

Comment:
Another standard pattern can be used here: `[DemoDemoClazz new]`. It is available for calling constructor without  parameters.

Reply:
thanks!